### PR TITLE
check minimum clang version

### PR DESCRIPTION
**Pull request by itaysk**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -501,6 +501,18 @@ func makeBPFObject(outFile string) error {
...
+	cmdVer := exec.Command(clang, "--version")
+	verOut, err := cmdVer.CombinedOutput()
+	if err != nil {
+		return err
+	}
+	// version output looks like: clang version 9.0.0-2 (tags/RELEASE_900/final)\n more lines...
+	eol := strings.Index(string(verOut), "\n")
+	verStr := string(verOut[14:eol])
+	verMajor, err := strconv.Atoi(strings.SplitN(verStr, ".", 2)[0])
+	if err != nil || verMajor < 7 {
```
Comment (yanivagman) [inline main.go:513]: I would separate this into error, and less than 7.
If and error detected - warn, but continue
Exit if major<7

File: main.go [hunk 2]
```diff
@@ -501,6 +501,18 @@ func makeBPFObject(outFile string) error {
...
+		return fmt.Errorf("required minimum clang version: 7")
```
Comment (yanivagman) [inline main.go:514]: Can you add the verStr as part of the output (something like: "detected version: xxx")

File: main.go (hunk 1 above)
Comment (itaysk) [reply main.go:513]: if `clang --version` returned an unrecognized format, you want to ignore?

Comment (yanivagman) [reply main.go:513]: The problem is that "clang --version" is not guranteed to keep its format between versions...
Do we want to fail if this is changed in new versions?

Comment (itaysk) [reply main.go:513]: I'm ok with both options, just wanted to clarify your request. you prefer to ignore?

Comment (yanivagman) [reply main.go:513]: I think so...

**Comment by itaysk** [comment]
@yanivagman I found a clang version that's not compatible, will send an update soon

**Review by yanivagman — APPROVED** [review]

**Pull request merged**